packages/client/src/actions/Actions.ts

```diff
@@ -34,90 +34,63 @@ import type {
   attachment: Attachment;
 }
+// THIS IS IT
+type BaseType<APD extends ActionParameterDefinition<any, any>> =
+  APD["type"] extends ObjectActionDataType<any, infer TTargetType> ?
+      | OsdkObjectFrom<TTargetType>
+      | OsdkObjectPrimaryKeyType<TTargetType>
+    : APD["type"] extends ObjectSetActionDataType<any, infer TTargetType>
+      ? ObjectSet<TTargetType>
+    : APD["type"] extends keyof OverrideWirePropertyTypes
+      ? OverrideWirePropertyTypes[APD["type"]]
+    : never;
 
-type OsdkActionParameterBaseType<
-  O extends OntologyDefinition<any>,
-  K extends ActionKeysFrom<O>,
-  P extends keyof O["actions"][K]["parameters"],
-> = ActionParameterTypeFrom<O, K, P> extends
-  ObjectActionDataType<infer TObjectName> ?
-    | OsdkObjectFrom<O["objects"][TObjectName]>
-    | OsdkObjectPrimaryKeyType<O["objects"][TObjectName]>
-  : ActionParameterTypeFrom<O, K, P> extends
-    ObjectSetActionDataType<infer TObjectName>
-    ? ObjectSet<ObjectOrInterfaceDefinitionFrom<O, TObjectName>>
-  : ActionParameterTypeFrom<O, K, P> extends keyof OverrideWirePropertyTypes
-    ? OverrideWirePropertyTypes[ActionParameterTypeFrom<O, K, P>]
-  : never;
-
-type OsdkActionParameterMaybeArrayType<
-  O extends OntologyDefinition<any>,
-  K extends ActionKeysFrom<O>,
-  P extends ActionParameterKeysFrom<O, K>,
-> = ActionParameterDefinitionFrom<O, K, P>["multiplicity"] extends true
-  ? Array<OsdkActionParameterBaseType<O, K, P>>
-  : OsdkActionParameterBaseType<O, K, P>;
+type MaybeArrayType<APD extends ActionParameterDefinition<any, any>> =
+  APD["multiplicity"] extends true ? Array<BaseType<APD>>
+    : BaseType<APD>;
 
-type OsdkActionParametersNotOptional<
-  O extends OntologyDefinition<any>,
-  K extends ActionKeysFrom<O>,
-> = {
-  [P in ActionParameterKeysFrom<O, K>]: OsdkActionParameterMaybeArrayType<
-    O,
-    K,
-    P
-  >;
+type NotOptionalParams<X extends ActionParametersDefinition> = {
+  [P in keyof X]: MaybeArrayType<X[P]>;
 };
 
-type OsdkActionParameters<
-  O extends OntologyDefinition<any>,
-  K extends ActionKeysFrom<O>,
-> = NullableProps<ActionDefinitionFrom<O, K>["parameters"]> extends never
-  ? OsdkActionParametersNotOptional<O, K>
-  : PartialByNotStrict<
-    OsdkActionParametersNotOptional<O, K>,
-    NullableProps<ActionDefinitionFrom<O, K>["parameters"]>
-  >;
+export type OsdkActionParameters<
+  X extends ActionParametersDefinition,
+> = NullableProps<X> extends never ? NotOptionalParams<X>
+  : PartialByNotStrict<NotOptionalParams<X>, NullableProps<X>>;
 
 export type Actions<O extends OntologyDefinition<any>> = {
-  [K in ActionKeysFrom<O>]: <OP extends ApplyActionOptions>(
-    args: NOOP<OsdkActionParameters<O, K>>,
-    options?: OP,
-  ) => Promise<ActionReturnTypeForOptions<OP>>;
+  [K in keyof O["actions"]]:
+    NonNullable<O["actions"][K]["__OsdkActionType"]> extends never
+      ? ActionSignature<O["actions"][K]["parameters"]>
+      : NonNullable<O["actions"][K]["__OsdkActionType"]>;
```

Comment:
Use the marker version if its specified, otherwise use the fallback. This gets us description through the marker.